docs/gene_sets.md

```diff
@@ -0,0 +1,122 @@
+|---------------|----------------------|-------------|------------------|------------------|-------------------------|
+| club.cell     | description          | CCKAR       | description      | Pubmed ID XYZ123 | Primary Pubmed ID       |
+| club.cell     |                      | SCGB3A2     | description      | Pubmed ID ABC456 | Primary Pubmed ID       |
+| club.cell     |                      | CYP2F2      | description      | Pubmed ID DCF678 | Primary Pubmed ID       |
+| macrophage    | description          | CD68        | description      |                  |                         |
+| macrophage    |                      | CD163       | description      |                  |                         |
+
+
+## Mandatory Header
+
+The first row MUST contain header columns using reserved names in the following order:
+
+* `gene_set_name`
+* `gene_set_description`
+* `gene_symbol`
+* `gene_description`
+
+Publishers MAY include additional header columns. It is RECOMMENDED that these custom columns observe the same self-documenting style - `column_name` and `column_description` for easier comprehension by data consumers.
+
+Only the values of `gene_set_name` and `gene_set_description` are presented to data consumers in the portal UX. 
+
+## Rows
+
+Each subsequent row describes a gene in a gene set.
+<br><br>
+
+### `gene_set_name`
+
+---
+
+The `gene_set_name` column MUST contain a value.
+
+If the `gene_set_name` is missing, validation MUST display a warning and fail the upload. This is illustrated by **~~?~~** in the example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description |
+|---------------|----------------------|-------------|------------------|
+| club.cell     | description          | CCKAR       | description      |
+|     **~~?~~** |                      | SCGB3A2     | description      |
+| club.cell     |                      | CYP2F2      | description      |
+
+
+All `gene_symbol(s)` for a `gene_set_name` MUST be on contiguous rows; otherwise, an out-of-order row is interpreted as a duplicate `gene_set_name`. In this case, validation MUST display a warning and fail the upload. This is illustrated by **~~club.cell~~** in the example:
+
+---
+| gene_set_name | gene_set_description | gene_symbol | gene_description |
+|---------------|----------------------|-------------|------------------|
+| club.cell     | description          | CCKAR       | description      |
+| club.cell     |                      | SCGB3A2     | description      |
+| club.cell     |                      | CYP2F2      | description      |
+| macrophage    | description          | CD68        | description      |
+| **~~club.cell~~**     |                      | CD163       | description      |
+
+
+When new gene sets are being added to a data collection on the portal, validation MUST detect `gene_set_name` collisions with current gene sets in the collection, display a warning, and fail the upload.  <br><br>
```

Comment:
What is the process by which a single gene set in a collection can be updated (e.g. adding a gene to a gene set)?

Reply:
In a *private* collection, this would require the curator to:

1. Delete the single gene set
2. Upload a gene set file containing the updated single gene set
3. Re-link the single gene set to the datasets (if previously linked)

In the revision of a *public* collection which is in refinement, it currently follows the same model as private. We only support replace for datasets due to the requirement to maintain their CXG permalinks. 

Are you suggesting that the portal interpret collisions as `replace the gene set` in the collection? Based on your wrangling experience how often do you see updates to gene sets?

Reply:
I wasn't suggesting anything, just seeking clarification.
Still trying to wrap my head around how the information we're capturing might translate to the gene sets you're collecting. But right now, I don't anticipate updates occurring frequently enough to warrant effort toward optimization. The process you described makes sense.